### DISCUSSION TITLE
cumulative dynamic AUC, error with NaN values.

### DISCUSSION THREAD

**Discussion by farrimoh** [sebp/scikit-survival#337] [Q&A]
When using cumulative_dynamic_auc function to calculate cumulative dynamic AUC, I get an error that the data should not have NaN values. (https://scikit-survival.readthedocs.io/en/stable/api/generated/sksurv.metrics.cumulative_dynamic_auc.html)
I set the 2d array such that the positive data doesn't have nan values (for the cumulative sensitivity).
However the negative data will have NaN values because trajectories don't have the same length.
But the implementation of the function is such that it does not accept NaN values in the 2d array.
Is there a fix for this problem?

**Comment by sebp** [comment]
I'm not sure I fully understand what you are asking.

Could you provide a minimal working example that illustrates the problem?

**Comment by farrimoh** [comment]
Thanks for your reply.
Here is an example of time dependent predictions for positive and negative. I want to calculate cumulative/dynamic auc at each hour>0 for this data.

![Data](https://user-images.githubusercontent.com/41919898/216126324-197ab02c-6820-44d7-96f2-9fe00083b685.png)
The implementation requires the 2d array that is passed to it, to not have any nan values. This makes sense for positive, since it is used for calculating cumulative sensitivity, so I use df.loc[dff['Label']==1,:].fillna(method='ffill',axis=1) to fill all the nan values with the last value of positive. 
But not for negative, negative data is used for dynamic sensitivity, so I would expect the function to accept the 2d array from the below data, and when calculating auc at time t0, it should resize the data by dropping negative data that is all nan after that time=t0.

![After_fillna](https://user-images.githubusercontent.com/41919898/216128014-239de3f8-4a98-48bb-8f05-7a5159a26814.png)

**Reply by sebp** [reply]
What do you mean with "positive" and "negative"? Each column should correspond to the risk scores for that particular time point `t_i`. Then "positive" and "negative" are determined based on whether a sample's observed event time is before or after that time point `t_i` (while accounting for right-censoring). If you pass NaN as risk score for some time point, I can't imagine a sensible way to sort samples by risk score.

**Reply by farrimoh** [reply]
If you cannot pass NaN values, how do you account for censoring?

**Reply by sebp** [reply]
Censoring is accounted for by inverse probability of censoring weights in the formula for the AUC, i.e. the omegas in the formula mentioned in the documentation: https://scikit-survival.readthedocs.io/en/stable/api/generated/sksurv.metrics.cumulative_dynamic_auc.html

Such a inverse probability of censoring weighting is very common to deal with censoring.
